examples/event-sources/gcp-pubsub-workflow-identity.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: argoproj.io/v1alpha1
+kind: EventSource
+metadata:
+  name: gcp-pubsub-event-source
+spec:
+  type: pubsub
+  pubSub:
+    example:
+      # jsonBody specifies that all event body payload coming from this
+      # source will be JSON
+      jsonBody: true
+      # id of your project
+      projectID: argo-events-XXXXX
+      # (optional) id of project for topic, same as projectID by default
+      # topicProjectID: "project-id"
+      # topic name
+      topic: test
+      # Empty credentials file when using Workflow Identity
+      credentialsFile: ""
+      # If enableWorkflowIdentity is true, the projects uses Workflow Identity for authentication
+      enableWorkflowIdentity: true
```

Comment:
Can we do something like:

if len(credentialsFile) == 0 {
  // consider it is using workload identity
} else {
  //
}

Reply:
That would make it simpler, but I think it's helpful to include the enableWorkflowIdentity flag to inform users that Workflow Identity is an option

Reply:
I think informing the users that Workflow Identity is supported in the doc might be better than in the code.